dtos/friendships/friendship.dto.ts

```diff
@@ -25,7 +25,6 @@ export class FriendshipDto {
     description: 'The id of the second user',
   })
   @IsInt()
-  @IsPositive()
```

Comment:
Was this intentional? I think the user id will still need to be positive

Reply:
Yes it is. We come to this solution with Nina, in order to create an easy system for adding friends.

We had the following problem: to add friends in the frontend, the best way for the user is to search for these friends by username.
The simple solution  was to create a route, POST friendships/username.
This works by creating a new friendship with the sourceId (id of the friend sending the request) and the username of the target friend. Then in createFromUsername(), the username is used to find the id of the target. createFromUsername() uses createfriendshipDto.

But when we search for a user by username, we don't have the id of the target, so we set it to 0. If we keep @IsPositive(), the targetId cannot be set to 0.

And we didn't find case where targetId doesn't have @IsPositive() and that creates problems.